addons/api/addon/transforms/object.js

```diff
@@ -12,20 +12,16 @@ export default class ObjectTransform extends Transform {
    * @returns {object}
    */
   deserialize(serialized) {
-    if (Object.keys(serialized).length) {
-      const obj = new TrackedObject(serialized);
-      return obj;
-    } else {
-      return {};
-    }
+    const obj = new TrackedObject(serialized || {});
+    return obj;
   }
 
   /**
    * @param deserialized
    * @returns {object}
    */
   serialize(deserialized) {
-    if (Object.keys(deserialized).length) {
+    if (deserialized && Object.keys(deserialized).length) {
```

Comment:
If it's null it might be fine to just return null? Do we need to check at all since I assume the API would want null if there's nothing rather than an empty object

Reply:
yeah that should be fine, I'll update